mars/services/scheduling/worker/execution.py

```diff
@@ -274,12 +247,93 @@ async def internal_run_subtask(self, subtask: Subtask, band_name: str):
                                                 subtask.session_id)
             yield task_api.set_subtask_result(subtask_info.result)
 
+    async def _retry_run_subtask(self, subtask: Subtask, band_name: str, subtask_api: SubtaskAPI):
```

Comment:
Need refactoring. Using nested `asynccontextmanager` is not clear. Handling exceptions with `issubclass` is also not Pythonic. A recommended way is to separate run and retry into two separate methods.

Reply:
Can I move the `asynccontextmanager` to the module level? I think it is much clearer than `try ... except ...`, and is much better than the async generator. Using an async generator in a method has different meanings than in the actor method. The latter one will release the actor lock.